designs/no-schedule-taint.md

```diff
@@ -0,0 +1,37 @@
+
+Cluster Autoscaler [uses their own `NoSchedule` taint](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/utils/taints/taints.go#L39-L42) to ensure that pods will not schedule to the nodes being deprovisioned. [By default](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/main.go#L216-L217), CAS will evict daemonset pods for non-empty nodes and will also respect a pod [annotation](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-can-i-enabledisable-eviction-for-a-specific-daemonset) on each daemonset pod to know if Karpenter should evict the pod.
+
+🔑 Karpenter determines if it should evict a pod through its tolerations. Karpenter could consider adding in a configuration surface similar to CAS to allow evicting pods owned by a daemonset. I recommend that we make this taint change during v1beta1 with ample documentation, and add the configuration surface in as a follow up if requested.
+
+### SIG-Autoscaling Alignment
+
+Karpenter is currently in process of aligning with SIG-Autoscaling for a common experience. As part of alignment, all API groups for both projects are proposed to be renamed to `node-lifecycle.kubernetes.io`.  After this change, users migrating to `v1beta1` will need to change their daemonsets to tolerate the new Karpenter taint if they want their daemonsets not evicted. If the API group is changed as part of alignment, these API groups will change once more, requiring users to change their tolerations on their workloads once again, introducing more churn. As this alignment will be a breaking change for both CAS and Karpenter, this change is inevitable, and should be done when Karpenter once again at a future time, potentially when graduating APIs to `v1`.
+
+🔑 Every change Karpenter makes to an API group causes cluster administrators to enforce application level changes. The `v1beta1` migration initially included a change for AWS API groups of `karpenter.k8s.aws->compute.k8s.aws`, where it was decided to keep as is, since the SIG-Autoscaling alignment could have changed the API group once again to `node-lifecycle.kubernetes.io`. We should change the taint now as a breaking change to behavior specific to `v1beta1`, and hold the next API group change until v1.
+
+### How severe is this issue currently?
+
+At the moment, as far as we know, no user has reported this issue, likely since its effect and recovery are mitigated by how Karpenter deprovisions serially, executing one action at a time. If Karpenter crashes/restarts while executing a command, only one command’s worth of candidate nodes will be left cordoned. Since expiration, drift, and consolidation all have a consistent ordering, it’s then likely that each cordoned node will get chosen for a deprovisioning action soon after recovery.
```

Comment:
```suggestion
At the moment, no user has reported this issue, likely since its effect and recovery are mitigated by how Karpenter deprovisions serially, executing one action at a time. If Karpenter crashes/restarts while executing a command, only one command’s worth of candidate nodes will be left cordoned. Since expiration, drift, and consolidation all have a consistent ordering, it’s then likely that each cordoned node will get chosen for a deprovisioning action soon after recovery.
```

I'd drop this since it feels like you're getting into [weasel word](https://en.wikipedia.org/wiki/Weasel_word) territory.